core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -194,7 +194,8 @@ public List<TableIdentifier> listTables(Namespace namespace) {
   public void renameTable(TableIdentifier from, TableIdentifier to) {
     int updatedRecords = execute(
         err -> {
-          if (err instanceof SQLIntegrityConstraintViolationException) {
+          if (err instanceof SQLIntegrityConstraintViolationException ||
+              err.getMessage() != null && err.getMessage().contains("UNIQUE constraint failed")) {
```

Comment:
I don't think this test is going to hold up for most JDBC databases.

There's a SQLCode on the error, but it was SQLIte specific in the test runs.

Reply:
We looked into this and there isn't a good way to do it. In `JdbcTableOperations`, we do the same thing but check for just "constraint failed". I pasted some links for background in my other comment.

Reply:
Updated to use the referenced code (with the same comment).